$OpenBSD: patch-nss_coreconf_OpenBSD_mk,v 1.5 2019/05/28 17:00:54 landry Exp $

- respect SHARED_LIBS
- use the version script to avoid symbol clashes with libcrypto
cf https://hg.mozilla.org/projects/nss/rev/383fbab6baf5ff2a694aae4dac6da7ffc2dd5342

Index: nss/coreconf/OpenBSD.mk
--- nss/coreconf/OpenBSD.mk.orig
+++ nss/coreconf/OpenBSD.mk
@@ -24,7 +24,7 @@ OS_LIBS			+= -pthread
 DSO_LDOPTS		+= -pthread
 endif
 
-DLL_SUFFIX		= so.1.0
+DLL_SUFFIX		= so.${SO_VERSION}
 
 OS_CFLAGS		= $(DSO_CFLAGS) $(OS_REL_CFLAGS) -Wall -Wno-switch -pipe -DOPENBSD
 
@@ -36,6 +36,11 @@ DSO_CFLAGS		= -fPIC -DPIC
 DSO_LDOPTS		= -shared -fPIC -Wl,-soname,lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
 
 MKSHLIB			= $(CC) $(DSO_LDOPTS)
+ifdef MAPFILE
+	MKSHLIB += -Wl,--version-script,$(MAPFILE)
+endif
+PROCESS_MAP_FILE = grep -v ';-' $< | \
+        sed -e 's,;+,,' -e 's; DATA ;;' -e 's,;;,,' -e 's,;.*,;,' > $@
 
 USE_SYSTEM_ZLIB		= 1
 ZLIB_LIBS		= -lz
